src/main/java/org/jboss/pnc/bifrost/endpoint/LogUpload.java

```diff
@@ -93,6 +100,29 @@ public String uploadFinalLog(@Valid FinalLogUpload logUpload, @Context HttpHeade
         return "ok";
     }
 
+    @Path("/{processContext}/delete")
+    @DELETE
+    @RolesAllowed("**")
+    @Transactional // FIXME change to specific allowed roles
+    public Response deleteFinalLog(@PathParam("processContext") String processContext) {
+        // parse process context
+        Long processContextLong = idConverter.convert(processContext);
+        if (processContextLong == null) {
+            throw new BadRequestException("Process context " + processContext + "is not a number nor a Build process.");
+        }
+
+        List<LogEntry> logEntries = LogEntry.list("processContext", processContextLong);
+        if (logEntries.stream().noneMatch(LogEntry::getTemporary)) {
```

Comment:
I think it should by `!allMatch(getTemporary)`. This way if just one is temporary, then it will delete it even if the rest is for some reason persistent. This should not happen, buuut.....

Reply:
`!allMatch()` is equal to `noneMatch()`

Reply:
It is not.
```
elements  allMatch  !allMatch  noneMatch
T,T       T         F          F
T,F       F         T          F
F,F       F         T          T
```